pallets/automation-time/src/lib.rs

```diff
@@ -787,52 +798,62 @@ pub mod pallet {
 		/// 
 		/// A task has been completed if all execution times exist in the past and/or are in the current time slot.
 		/// This means there a no more execution times to be run.
-		fn is_completed_task(execution_times: BoundedVec<UnixTime, T::MaxRecurringTimes>) -> bool {
+		fn is_completed_task(execution_times: Vec<UnixTime>) -> bool {
 			let current_time_slot = match Self::get_current_time_slot() {
 				Ok(time_slot) => time_slot,
-				Err(_) => return false, // fix error
+				// This will only occur for the first block in the chain.
+				Err(_) => return false,
 			};
 
 			let scheduled_times = Self::clean_execution_times_vector(execution_times);
-			return *scheduled_times.last().unwrap() <= current_time_slot;
+			let is_complete = *scheduled_times.last().unwrap() <= current_time_slot;
+			return is_complete;
 		}
 
 		/// Removes the task of the provided task_id and all scheduled tasks, including those in the task queue.
 		///
-		fn remove_task(task_id: T::Hash, task: Task<T>) {
-			let current_time_slot = match Self::get_current_time_slot() {
-				Ok(time_slot) => time_slot,
-				Err(_) => return, // fix error
-			};
+		fn remove_task(task_id: T::Hash, task: Task<T>) -> Result<(), Error<T>> {
 			let mut found_task: bool = false;
-			let execution_times = Self::clean_execution_times_vector(task.execution_times);
+			let current_time_slot = Self::get_current_time_slot()?;
+			let execution_times = Self::clean_execution_times_vector(task.execution_times.to_vec());
 
-			for time in execution_times.iter().rev() {
-				if *time < current_time_slot { break; }
-				match Self::get_scheduled_tasks(*time) {
-					Some(mut task_ids) =>
-						for i in 0..task_ids.len() {
-							if task_ids[i] == task_id {
-								if task_ids.len() == 1 {
-									<ScheduledTasks<T>>::remove(*time);
-								} else {
-									task_ids.remove(i);
-									<ScheduledTasks<T>>::insert(*time, task_ids);
-								}
+			if let Some((last_time_slot, _)) = Self::get_last_slot() {
+				for time in execution_times.iter().rev() {
+					// Execution time is less than current time slot and in the past.  No more execution times need to be removed.
+					if *time < current_time_slot { 
+						break
+					}
+					// Execution time is equal to last time slot and task queue should be checked for task id.
+					// After checking task queue no other execution times need to be removed.
+					if *time == last_time_slot { 
```

Comment:
why do we need to do this check with last_time_slot? if we just comb through the task_queue, it should be find the task_id if it's in there. and if not, then just need to comb through the different scheduled times i think?

also i worry about edge case during the start of the hour when the last_time_slot changes, if someone tries to remove then, before the first block builds and changes the last_time_slot to something new, if something weird happens there and we fail to remove from task queue